WooCommerce/Classes/Authentication/Navigation Exceptions/WrongAccountErrorViewModel.swift

```diff
@@ -0,0 +1,138 @@
+/// an error when Jetpack is not installed or is not connected
+struct WrongAccountErrorViewModel: ULAccountMismatchViewModel {
+    private let siteURL: String
+    private let defaultAccount: Account?
+
+    init(siteURL: String?, sessionManager: SessionManagerProtocol =  ServiceLocator.stores.sessionManager) {
+        self.siteURL = siteURL ?? Localization.yourSite
+        self.defaultAccount = sessionManager.defaultAccount
+    }
+
+    // MARK: - Data and configuration
+    var userEmail: String {
+        guard let account = defaultAccount else {
+            DDLogWarn("⚠️ Present account mismatch UI for \(siteURL) without a default account")
+
+            return ""
+        }
+        return account.email
+    }
+
+    var userName: String {
+        guard let account = defaultAccount else {
+            DDLogWarn("⚠️ Present account mismatch UI for \(siteURL) without a default account")
+
+            return ""
+        }
+
+        return account.displayName
+    }
+
+    var signedInText: String {
+        guard let account = defaultAccount else {
+            DDLogWarn("⚠️ Present account mismatch UI for \(siteURL) without a default display account")
+
+            return ""
+        }
+
+        return String.localizedStringWithFormat(Localization.signedInMessageFormat,
+                                                account.username)
+    }
+
+    let logOutTitle: String = Localization.wrongAccountMessage
+
+    let logOutButtonTitle: String = Localization.logOutButtonTitle
+
+    let image: UIImage = .errorImage
+
+    var text: NSAttributedString {
+        let font: UIFont = .body
+        let boldFont: UIFont = font.bold
+
+        let boldSiteAddress = NSAttributedString(string: siteURL.trimHTTPScheme(),
+                                                           attributes: [.font: boldFont])
+        let message = NSMutableAttributedString(string: Localization.errorMessage)
+
+        message.replaceFirstOccurrence(of: "%@", with: boldSiteAddress)
+
+        return message
+    }
+
+    let isAuxiliaryButtonHidden = false
+
+    let auxiliaryButtonTitle = Localization.findYourConnectedEmail
+
+    let primaryButtonTitle = Localization.primaryButtonTitle
+
+    // MARK: - Actions
+    func didTapPrimaryButton(in viewController: UIViewController?) {
+        guard let navigationController = viewController?.navigationController else {
+            return
+        }
+
+        let storePicker = StorePickerViewController()
+        storePicker.configuration = .listStores
+        //storePickerCoordinator?.onDismiss = onDismiss
+        navigationController.pushViewController(storePicker, animated: true)
+    }
+
+    func didTapAuxiliaryButton(in viewController: UIViewController?) {
+        let fancyAlert = FancyAlertViewController.makeNeedHelpFindingEmailAlertController()
+        fancyAlert.modalPresentationStyle = .custom
+        fancyAlert.transitioningDelegate = AppDelegate.shared.tabBarController
```

Comment:
This feels a bit drastic, but I'm not sure why is it needed 🤔 
Could you explain it a little bit more?

Reply:
We need to set the transitioningDelegate otherwise the fancy alert does not grab out the background. I presume what's called you attention is pulling the tabBarController from the AppDelegate directly? 

If that's the case, I guess we could pull the tabBarController from the UIViewController itself, but then we'd need to downcast to MainTabBarController, which I am not sure makes things better. Also, it's the pattern used elsewhere.

I am not sure if that answers you question though.

Reply:
Thanks for the explanation, yes it was due to pulling the "tabBarController from the AppDelegate directly"
I don't have a better suggestion at the moment so let's leave it like that if you prefer.